src/main/java/de/rub/nds/modifiablevariable/util/ArrayConverter.java

```diff
@@ -13,43 +13,89 @@
 import java.util.List;
 
 /** A utility class to handle arrays and array conversions */
-public class ArrayConverter {
+public final class ArrayConverter {
+
+    private ArrayConverter() {
+        super();
+    }
 
     /**
      * Takes a long value and converts it to 8 bytes (needed for example to convert SQN numbers in
      * TLS records)
      *
-     * @param l long value
+     * @param value long value
      * @return long represented by 8 bytes
      */
-    public static byte[] longToUint64Bytes(long l) {
+    public static byte[] longToEightBytes(long value) {
         byte[] result = new byte[8];
-        result[0] = (byte) (l >>> 56);
-        result[1] = (byte) (l >>> 48);
-        result[2] = (byte) (l >>> 40);
-        result[3] = (byte) (l >>> 32);
-        result[4] = (byte) (l >>> 24);
-        result[5] = (byte) (l >>> 16);
-        result[6] = (byte) (l >>> 8);
-        result[7] = (byte) (l);
+        result[0] = (byte) (value >>> 56);
+        result[1] = (byte) (value >>> 48);
+        result[2] = (byte) (value >>> 40);
+        result[3] = (byte) (value >>> 32);
+        result[4] = (byte) (value >>> 24);
+        result[5] = (byte) (value >>> 16);
+        result[6] = (byte) (value >>> 8);
+        result[7] = (byte) value;
         return result;
     }
 
+    /** Note: This will truncate the long */
+    public static byte[] longToSixBytes(long value) {
+        byte[] output = new byte[6];
+        output[0] = (byte) (value >>> 40);
+        output[1] = (byte) (value >>> 32);
+        output[2] = (byte) (value >>> 24);
+        output[3] = (byte) (value >>> 16);
+        output[4] = (byte) (value >>> 8);
+        output[5] = (byte) value;
+        return output;
+    }
+
+    public static byte[] longToUint64Bytes(long value) {
+        return longToEightBytes(value);
+    }
+
+    public static byte[] longToUint48Bytes(long value) {
+        return longToSixBytes(value);
+    }
+
+    public static byte[] longToUint32Bytes(long value) {
+        return intToFourBytes((int) value);
+    }
```

Comment:
why do we have double definitions?

Reply:
these are the old names I added these back in so, you do not have to change each project using these functions.

Reply:
reverted

Reply:
for uint64 i can see that its wrong - cause longs cannot represent a full uint64 - the others can go. In case you do not want to outright remove - you should @Deprecate the old functions